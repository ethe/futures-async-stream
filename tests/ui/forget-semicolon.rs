// SPDX-License-Identifier: Apache-2.0 OR MIT

#![feature(coroutines)]

use futures_async_stream::{stream, try_stream};

#[stream(item = ())]
async fn foo() {
    yield;
    Some(()) //~ ERROR mismatched types
}

#[try_stream(ok = (), error = ())]
async fn bar() {
    yield;
    Some(()) //~ ERROR mismatched types
}

fn main() {}
