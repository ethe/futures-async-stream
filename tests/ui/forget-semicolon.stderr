error[E0308]: mismatched types
 --> tests/ui/forget-semicolon.rs:8:5
  |
8 |     Some(()) //~ ERROR mismatched types
  |     ^^^^^^^^- help: consider using a semicolon here: `;`
  |     |
  |     expected `()`, found enum `Option`
  |
  = note: expected unit type `()`
                  found enum `Option<()>`

error[E0308]: mismatched types
  --> tests/ui/forget-semicolon.rs:14:5
   |
14 |     Some(()) //~ ERROR mismatched types
   |     ^^^^^^^^- help: consider using a semicolon here: `;`
   |     |
   |     expected `()`, found enum `Option`
   |
   = note: expected unit type `()`
                   found enum `Option<()>`
